007-OFEP-transaction-context-propagation.md

```diff
@@ -0,0 +1,91 @@
+## Transaction Context Propagation
+
+## State: DRAFTING
+
+Evaluation context is a container for all contextual data that can be used used to conditionally control the value returned from a flag evaluation. Traditionally, a developer would be responsible for explicitly defining evaluation context during flag evaluation.
+
+Transaction context, a subset of the evaluation context, is a container for transaction specific data (e.g. user id, user agent, IP) that lives for the duration of a transaction. With transaction context propagation, a developer can set transaction context where it's convenient (e.g. an auth service) and have it persist for the length of a transaction (e.g. a request or thread). OpenFeature will then merge transaction context with evaluation context before flag evaluation.
+
+## Background
+
+Many languages provide a mechanism for storing data for the length of a single transaction. This can be used to store transaction context that can be automatically merged with the evaluation context before flag evaluation. The implementation will vary by technology. For example, Java may use [ThreadLocal][thread-local], Golang may use [Go Context][go-context], and Node may use [async hooks][async-hooks].
+
+## Proposal
+
+This proposal introduces a way to register a transaction context propagator to the global OpenFeature API. Once registered, transaction context can be defined at any point in the transaction. When a flag is evaluated, the transaction context is merged with evaluation context based on the merge order defined below.
```

Comment:
How does a developer set the transaction context at any point in the transaction?

Reply:
It would depend on the technology but there's an example using NodeJS later in the proposal.